### PR TITLE
Make MirrorFilter.BackendRef required

### PR DESCRIPTION
**Pull request by howardjohn**
/kind cleanup

/kind api-change

    Currently this is optional. I cannot see any reason why it would not be
    required, as the whole point of mirroring is the request goes to some
    backend?
Does this PR introduce a user-facing change?:

```notes
* Make `MirrorFilter.BackendRef` a required field when the mirror filter is used
```

### CODE REVIEW

File: apis/v1alpha2/httproute_types.go [hunk 1]
```diff
@@ -722,9 +722,7 @@ type HTTPRequestMirrorFilter struct {
 	//
 	// Support: Extended for Kubernetes Service
 	// Support: Custom for any other resource
-	//
-	// +optional
-	BackendRef *BackendObjectReference `json:"backendRef,omitempty"`
+	BackendRef BackendObjectReference `json:"backendRef"`
```
Comment (hbagdi) [inline apis/v1alpha2/httproute_types.go:725]: CI failing because tests are not compiling.

**Comment by hbagdi** [comment]
LGTM from an API perspective.

File: apis/v1alpha2/httproute_types.go (hunk 1 above)
Comment (howardjohn) [reply apis/v1alpha2/httproute_types.go:725]: thanks, should be good now

**Comment by hbagdi** [comment]
@howardjohn Can you update the release notes section of the PR description so that it makes sense to controller authors and end-users?

/lgtm

**Comment by robscott** [comment]
Thanks!

/approve

**Pull request merged**